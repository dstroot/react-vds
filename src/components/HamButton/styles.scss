@charset "UTF-8";
/*!
 * Hamburgers
 * @description Tasty CSS-animated hamburgers
 * @author Jonathan Suh @jonsuh
 * @site https://jonsuh.com/hamburgers
 * @link https://github.com/jonsuh/hamburgers
 */

// Settings
// ==================================================
$hamburger-padding-x: 15px !default;
$hamburger-padding-y: 15px !default;
$hamburger-layer-width: 30px !default;
$hamburger-layer-height: 3px !default;
$hamburger-layer-spacing: 6px !default;
$hamburger-layer-color: #000 !default;
$hamburger-layer-border-radius: 4px !default;
$hamburger-hover-opacity: 0.7 !default;
$hamburger-active-layer-color: $hamburger-layer-color !default;
$hamburger-active-hover-opacity: $hamburger-hover-opacity !default;

// To use CSS filters as the hover effect instead of opacity,
// set $hamburger-hover-use-filter as true and
// change the value of $hamburger-hover-filter accordingly.
$hamburger-hover-use-filter: false !default;
$hamburger-hover-filter: opacity(50%) !default;
$hamburger-active-hover-filter: $hamburger-hover-filter !default;

// Types (Remove or comment out what you donâ€™t need)
// ==================================================
$hamburger-types: (
  // 3dx,
    // 3dx-r,
    // 3dy,
    // 3dy-r,
    // 3dxy,
    // 3dxy-r,
    // arrow,
    // arrow-r,
    // arrowalt,
    // arrowalt-r,
    // arrowturn,
    // arrowturn-r,
    // boring,
    // collapse,
    // collapse-r,
    // elastic,
    // elastic-r,
    // emphatic,
    // emphatic-r,
    // minus,
    // slider,
    // slider-r,
    // spin,
    // spin-r,
    // spring,
    // spring-r,
    // stand,
    // stand-r,
    squeeze,
  // vortex,
    // vortex-r
) !default;

// Base Hamburger (We need this)
// ==================================================
@import './scss/base';

// Hamburger types
// ==================================================
// @import './scss/types/3dx';
// @import './scss/types/3dx-r';
// @import './scss/types/3dy';
// @import './scss/types/3dy-r';
// @import './scss/types/3dxy';
// @import './scss/types/3dxy-r';
// @import './scss/types/arrow';
// @import './scss/types/arrow-r';
// @import './scss/types/arrowalt';
// @import './scss/types/arrowalt-r';
// @import './scss/types/arrowturn';
// @import './scss/types/arrowturn-r';
// @import './scss/types/boring';
// @import './scss/types/collapse';
// @import './scss/types/collapse-r';
// @import './scss/types/elastic';
// @import './scss/types/elastic-r';
// @import './scss/types/emphatic';
// @import './scss/types/emphatic-r';
// @import './scss/types/minus';
// @import './scss/types/slider';
// @import './scss/types/slider-r';
// @import './scss/types/spin';
// @import './scss/types/spin-r';
// @import './scss/types/spring';
// @import './scss/types/spring-r';
// @import './scss/types/stand';
// @import './scss/types/stand-r';
@import './scss/types/squeeze';
// @import './scss/types/vortex';
// @import './scss/types/vortex-r';
