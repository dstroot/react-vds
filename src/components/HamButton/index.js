import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';

/**
 * https://jonsuh.com/hamburgers/
 * Let's use CSS rather than JS to animate the burger menu
 *
 * `y add hamburgers`
 */

const HamButton = ({ toggler, clicked }) => {
  let classes = 'hamburger hamburger--squeeze';
  if (clicked) {
    // Trigger the active state by appending class name is-active
    classes = 'hamburger hamburger--squeeze is-active';
  }
  return (
    <div className={classes} onClick={toggler}>
      <span className="hamburger-box">
        <span className="hamburger-inner" />
      </span>
    </div>
  );
};

export default HamButton;

HamButton.propTypes = {
  /** Function to toggle button clicked state */
  toggler: PropTypes.func.isRequired,
  /** Should be set to "true" if the button has been clicked */
  clicked: PropTypes.bool.isRequired,
};
