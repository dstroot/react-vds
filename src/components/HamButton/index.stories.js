import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import HamButton from '../HamButton';

storiesOf('Pac Life|HamButton', module)
  .addParameters({
    info: {
      inline: true,
      header: false,
      text: `This supports markdown!
      
      ~~~js
      console.log("hello");
      ~~~
      `,
    },
  })
  .add('show HamButton unclicked', () => (
    <HamButton clicked={false} toggler={action('button-click')} />
  ))
  .add('show HamButton clicked', () => (
    <HamButton clicked={true} toggler={action('button-click')} />
  ));
