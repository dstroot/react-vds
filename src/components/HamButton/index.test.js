import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
import HamButton from '../HamButton';

let clicked = false;
const mockToggler = jest.fn(() => (clicked = true));

describe('HamButton', () => {
  // mount the component
  let mountedComponent;
  const getMountedComponent = () => {
    if (!mountedComponent) {
      mountedComponent = mount(
        <HamButton toggler={mockToggler} clicked={clicked} />
      );
    }
    return mountedComponent;
  };

  beforeEach(() => {
    mountedComponent = undefined;
    clicked = false;
  });

  it('it should render', () => {
    let tree = create(<HamButton toggler={mockToggler} clicked={false} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('it should render clicked', () => {
    let tree = create(<HamButton toggler={mockToggler} clicked={true} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('clicking should call toggler function', () => {
    let div = getMountedComponent()
      .find('div')
      .first();

    expect(div.hasClass('is-active')).toBe(false);

    // now click the button
    div.simulate('click');

    // The mock function is called once
    expect(mockToggler.mock.calls.length).toBe(1);
  });
});
