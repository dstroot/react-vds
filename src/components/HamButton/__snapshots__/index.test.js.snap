// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HamButton it should render 1`] = `
<div
  className="hamburger hamburger--squeeze"
  onClick={[MockFunction]}
>
  <span
    className="hamburger-box"
  >
    <span
      className="hamburger-inner"
    />
  </span>
</div>
`;

exports[`HamButton it should render clicked 1`] = `
<div
  className="hamburger hamburger--squeeze is-active"
  onClick={[MockFunction]}
>
  <span
    className="hamburger-box"
  >
    <span
      className="hamburger-inner"
    />
  </span>
</div>
`;
